pkg/operators/common/resourcecache.go

```diff
@@ -0,0 +1,137 @@
+// Copyright 2024 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package common
+
+import (
+	"sync"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+)
+
+const (
+	oldObjectTTL = 2 * time.Second
+)
+
+type oldResource[T any] struct {
+	deletionTimestamp time.Time
+	obj               T
+}
+
+type ResourceCache[T any, Key comparable] struct {
+	sync.Mutex
+	current map[Key]*T
+	old     map[Key]oldResource[*T]
+	toKey   func(*T) (Key, error)
+}
+
+func NewResourceCache[T any, Key comparable](toKey func(*T) (Key, error)) *ResourceCache[T, Key] {
+	return &ResourceCache[T, Key]{
+		current: make(map[Key]*T),
+		old:     make(map[Key]oldResource[*T]),
+		toKey:   toKey,
```

Comment:
Well this cache grew from the `KubeInvCache`. And there I wanted to avoid explicitly getting the key from a Pod or service.
But yes. This is somehow uncommon what I did here